### ISSUE TITLE
Compile failure with latest build (2017-08-03)

### ISSUE THREAD

**Issue by Bladewood** [endless-sky/endless-sky#2884]
```
||=== Build: Release in EndlessSky (compiler: GNU GCC Compiler) ===|
C:\Users\litot_000\Documents\GitHub\endless-sky\endless-sky\source\AI.cpp||In member function 'void AI::AskForHelp(Ship&, bool&, const Ship*)':|
C:\Users\litot_000\Documents\GitHub\endless-sky\endless-sky\source\AI.cpp|698|error: no matching function for call to 'std::vector<Ship*>::insert(std::vector<Ship*>::const_iterator, double, Ship*)'|
C:\Users\litot_000\Documents\GitHub\endless-sky\endless-sky\source\AI.cpp|698|note: candidates are:|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\vector.tcc|107|note: std::vector<_Tp, _Alloc>::iterator std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, const value_type&) [with _Tp = Ship*; _Alloc = std::allocator<Ship*>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<Ship**, std::vector<Ship*> >; typename std::_Vector_base<_Tp, _Alloc>::pointer = Ship**; std::vector<_Tp, _Alloc>::value_type = Ship*]|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\vector.tcc|107|note:   candidate expects 2 arguments, 3 provided|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|988|note: std::vector<_Tp, _Alloc>::iterator std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::value_type&&) [with _Tp = Ship*; _Alloc = std::allocator<Ship*>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<Ship**, std::vector<Ship*> >; typename std::_Vector_base<_Tp, _Alloc>::pointer = Ship**; std::vector<_Tp, _Alloc>::value_type = Ship*]|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|988|note:   candidate expects 2 arguments, 3 provided|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|1005|note: void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::initializer_list<_Tp>) [with _Tp = Ship*; _Alloc = std::allocator<Ship*>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<Ship**, std::vector<Ship*> >; typename std::_Vector_base<_Tp, _Alloc>::pointer = Ship**]|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|1005|note:   candidate expects 2 arguments, 3 provided|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|1023|note: void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type, const value_type&) [with _Tp = Ship*; _Alloc = std::allocator<Ship*>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<Ship**, std::vector<Ship*> >; typename std::_Vector_base<_Tp, _Alloc>::pointer = Ship**; std::vector<_Tp, _Alloc>::size_type = long long unsigned int; std::vector<_Tp, _Alloc>::value_type = Ship*]|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|1023|note:   no known conversion for argument 1 from 'std::vector<Ship*>::const_iterator {aka __gnu_cxx::__normal_iterator<Ship* const*, std::vector<Ship*> >}' to 'std::vector<Ship*>::iterator {aka __gnu_cxx::__normal_iterator<Ship**, std::vector<Ship*> >}'|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|1044|note: template<class _InputIterator, class> void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, _InputIterator, _InputIterator) [with _InputIterator = _InputIterator; <template-parameter-2-2> = <template-parameter-1-2>; _Tp = Ship*; _Alloc = std::allocator<Ship*>]|
C:\Program Files\mingw64\x86_64-w64-mingw32\include\c++\bits\stl_vector.h|1044|note:   template argument deduction/substitution failed:|
C:\Users\litot_000\Documents\GitHub\endless-sky\endless-sky\source\AI.cpp|698|note:   cannot convert 'canHelp.std::vector<_Tp, _Alloc>::cend<Ship*, std::allocator<Ship*> >()' (type 'std::vector<Ship*>::const_iterator {aka __gnu_cxx::__normal_iterator<Ship* const*, std::vector<Ship*> >}') to type 'std::vector<Ship*>::iterator {aka __gnu_cxx::__normal_iterator<Ship**, std::vector<Ship*> >}'|
||=== Build failed: 1 error(s), 0 warning(s) (0 minute(s), 5 second(s)) ===|
```

**Comment by tehhowch** [comment]
change `cend` to `end`

**Comment by tehhowch** [comment]
and also change the ship.IsDisabled() in Engine.cpp to be ship->IsDisabled()
I could have sworn I wrote `canHelp.end()` and not `canHelp.cend()`... but... not so according to my branch history.

**Comment by endless-sky** [comment]
Fixed by PR #2885.
